### PR TITLE
[noetic] Added bodies::Body::computeBoundingBox (oriented box version)

### PR DESCRIPTION
**Pull request by peci1**
This is a noetic version of #105. I basically just cherry-picked the commits from #105 and resolved merge conflicts.

I also figured out this PR breaks ABI by adding a virtual function, so I again made it fake virtual. It's a pity it will be such in both noetic and melodic, but there's not much more that can be done.

I also made use of the fact that FCL 0.6 is available in noetic and relayed AABB computation in `bodies::AABB::extendWithTransformedBox()` directly to FCL. This change is a bit off-topic, but does not hurt and offers to get potential future improvements from FCL. I made this change in a separate commit, so if it is not desired, I can get rid of it.

### CODE REVIEW

**Comment by peci1** [comment]
The FCL search logic in CMake might look complicated, but that's what I had to do in https://github.com/peci1/robot_body_filter/commit/db08750759d5dd93822451a7c2981368c6bd187b#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20a in order to get it correctly working on Melodic and Noetic on both Ubuntu and Debian.

**Review by rhaschke — APPROVED** [review]
Looks good to me. I'm fine to include FCL in general. I would even drop ABI compatibility and risk some release friction in favour of cleaner code.

**Comment by peci1** [comment]
> I would even drop ABI compatibility

Should I remove the ABI compatibility commit, then?

**Comment by rhaschke** [comment]
> Should I remove the ABI compatibility commit, then?

Yes, please.

**Comment by rhaschke** [comment]
@ros-planning/moveit-maintainers, this is awaiting the 2nd approval for more than a month...

File: package.xml
Comment (simonschmeisser) [inline package.xml:22]: does this include apply for ROS2 as well? or should it be `==melodic` else ...

File: src/aabb.cpp
Comment (simonschmeisser) [inline src/aabb.cpp]: I doubt it but there might be a `1.0.0` release at some point

**Comment by simonschmeisser** [comment]
Do we actually intend to backport this to `melodic-devel`? If not it could be simplified a lot by dropping support for fcl 0.5 ... or maybe not as the libfcl package in Ubuntu Focal is still 0.5.0 while the ros-noetic-fcl is at 0.6.1

File: package.xml
Comment (peci1) [reply package.xml:22]: no idea how ROS 2 handles this... I haven't dived into it yet... so maybe just melodic/noetic would be better

**Comment by peci1** [comment]
I saw that versions of FCL can range from 0.5 to 0.6 even on newer systems depending on many things, so I'd rather keep it compatible with both. And yes, I'd like to backport this to melodic.

File: package.xml
Comment (simonschmeisser) [reply package.xml:22]: Debian bullseye contains 0.6.1, debian sid 0.7.1(and therefore also in Ubuntu 22.04) and there was no ros based releases except for noetic so this sounds like the correct approach

**Review by simonschmeisser — APPROVED** [review]

File: src/aabb.cpp
Comment (peci1) [reply src/aabb.cpp]: I added the major version checks, too.

**Comment by simonschmeisser** [comment]
Looks good now!

Do you want to squash the second commit into the first before merging?

**Comment by peci1** [comment]
> Do you want to squash the second commit into the first before merging?

Do you mean to not have a commit where build is broken?

**Comment by simonschmeisser** [comment]
> > Do you want to squash the second commit into the first before merging?
> 
> Do you mean to not have a commit where build is broken?

Yes and because the commit is somewhat "boring" :wink: and looks like a fixup

**Comment by peci1** [comment]
Okay, squashed ;)

**Comment by peci1** [comment]
Great, thanks! Now there's #105 which could be used for the melodic backport. Should I help with it somehow?

**Pull request merged**
